notebooks/Functions.ipynb

```diff
@@ -0,0 +1,337 @@
+       " ('2016-12-28T08:02:00Z', 17.22428760018352, -3.037106627394605),\n",
+       " ('2016-12-28T08:22:00Z', 12.573087266115454, -4.576229517697452),\n",
+       " ('2016-12-28T08:41:00Z', 14.988097301535237, -5.45522128604442),\n",
+       " ('2016-12-28T09:02:00Z', 14.92, 2.7407595364917763e-15),\n",
+       " ('2016-12-28T09:22:00Z', 18.74089153982232, -3.3045248210016775),\n",
+       " ('2016-12-28T11:43:00Z', 15.955980700944723, -5.807502033669859)]"
+      ]
+     },
+     "execution_count": 21,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Import the Python math module\n",
+    "import math\n",
+    "\n",
+    "# Constant for converting degrees to radians\n",
+    "RPERD = math.pi/180\n",
+    "\n",
+    "# The list that will hold our u and v data\n",
+    "uvdata = []\n",
+    "\n",
+    "# Loop through observations pulling apart the time, speed and direction\n",
+    "# components of the observation.\n",
+    "\n",
+    "for time, speed, direction in data:\n",
+    "    u = -speed * math.sin(direction * RPERD)\n",
+    "    v = -speed * math.cos(direction * RPERD)\n",
+    "    uvdata.append((time, u, v))\n",
+    "\n",
+    "uvdata"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "ein.tags": [
+     "worksheet-0"
+    ],
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "source": [
+    "While this code works fine it can be **refactored**, a software engineering term to describe improving the design of existing code. In particular, we can pull the section that calculates the `u` and `v` vectors into its own reusable unit of code known as a **function**. Functions in Python are defined with the `def` keyword as illustrated with this pseudocode:\n",
+    "\n",
+    "```\n",
+    "def function_name( function_parameters ):\n",
+    "   \"function documentation string\"\n",
+    "   function statements usually involving the function parameters\n",
+    "   return some value or expression based on the function statements\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "ein.tags": [
+     "worksheet-0"
+    ],
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "source": [
+    "Armed with this knowledge, we can write a function called `uandv` to calculate the u and v vectors from wind speed and direction parameters:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "autoscroll": "json-false",
+    "collapsed": false,
+    "ein.tags": [
+     "worksheet-0"
+    ],
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def uandv(speed, direction):\n",
+    "    \"calculate u and v vectors from speed and direction\"\n",
+    "    u = -speed * math.sin(direction * RPERD)\n",
+    "    v = -speed * math.cos(direction * RPERD)\n",
+    "    # return the u and v tuple\n",
+    "    return (u, v)"
```

Comment:
Unless you feel passionately that the parentheses are important for pedagogical purposes, I'd remove them since they're unneeded.